enhancements/operator-dev-doc.md

```diff
@@ -0,0 +1,239 @@
+
+```console
+$ make help
+```
+
+This builds the binaries    
+```console
+$ make build
+```
+
+This runs unit tests    
+```console
+$ make test-unit
+```
+
+This will run verify-gofmt, verify-govet, verify-bindata (if applicable), verify-codegen     
+There may be other verify targets added to individual Makefiles, also.     
+```console
+$ make verify
+```
+
+This will update-bindata (if applicable), update-codegen, update-gofmt    
+```console
+$ make update
+```
+
+If you have exported your KUBECONFIG to point to a running cluster, you can run    
+the end-to-end tests that live in the repository (that would run in CI with e2e-aws-operator)    
+```console
+$ make test-e2e
+```
+
+If the repository utilizes glide for dependency management, you can update dependencies with    
+```console
+$ make update-deps
+```
+
+If the repository uses gomod for dependency management, [this doc](https://docs.google.com/document/d/1WIxyDRbXIBCihLAYRFGVkET5dM7WOQkBvnuUf2DfnL0/edit?usp=sharing) is useful.
+
+### How can I test changes to an OpenShift operator/operand/release component? 
+
+When developing on OpenShift, at some point you'll want to test your changes in a running cluster.  Any component image in a release payload
+can be substituted for a locally built image.  You can test your changes in an OpenShift cluster like so:     
+```
+NOTE:
+You'll need to login to the quay registry (or docker.io) to push images.
+Also, quay.io by default sets new repositories to private.  You'll need to
+visit quay.io and change settings to public for any new image repos you push.
+This is to allow OpenShift to pull your image.
+```
+
+#### OPTION A)
+#### Build a new release payload and launch a cluster with that release
+
+For this example I'll start with the release image     
+`registry.svc.ci.openshift.org/ocp/release:4.2`     
+and test a change to the `github.com/openshift/openshift-apiserver` repository.     
+
+1. Build the image and push it to a registry (quay.io, docker.io)
+*Dockerfile name varies per repository, this example used Dockerfile.rhel, but it may have a different name*
+```
+$ cd local/path/to/openshift-apiserver
+$ buildah bud -f path/to/Dockerfile.rhel -t quay.io/yourname/openshift-apiserver:test .
+$ buildah push quay.io/yourname/openshift-apiserver:test
+```
+    
+2. Assemble a release payload with your test image and push it to a registry    
+Get the name of the image (`openshift-apiserver`) you want to substitute:
+```
+$ oc adm release info registry.svc.ci.openshift.org/ocp/release:4.2
+```
+
+This command will assemble a release payload incorporating your test image _and_ will push it to the quay.io repository.    
+Be sure to set this repository in quay.io as `public`.    
+```
+$ oc adm release new --from-release registry.svc.ci.openshift.org/ocp/release:4.2 \
+  openshift-apiserver=quay.io/yourname/openshift-apiserver:test \
+  --to-image quay.io/yourname/release:test
```

Comment:
When I ran this command the first time, I saw the following error:

```
info: Found 98 images in release
error: unable to read image quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:f33d299f2ba826d8da0d9ad41a59aa1b5363c425f292d42006a6001fcad08f8c: unauthorized: access to the requested resource is not authorized
```

@abhinavdahiya pointed out that the necessary credentials for quay.io are provided in the pull secret sourced from try.openshift.com. It might be worth mentioning that local docker configuration should include the pull secret (perhaps with instructions on how to merge the pull secret to an existing config e.g. `jq -s 'add' pull-secret.json ~/.docker/config.json`).

Reply:
right, will do

Reply:
@marun I fiddled with `jq -s 'add' ` and did not get the result required.  If you have better instructions than `Add the quay auth from the pull-secret to ~/.docker/config.json` with a sample o f the resulting config.json : ) lemme know, thanks